pymongo/mongo_client.py

```diff
@@ -2084,16 +2086,16 @@ def __init__(self, client, server, session):
         # completes then the error's generation number is the generation
         # of the pool at the time the connection attempt was started."
         self.sock_generation = server.pool.gen.get_overall()
-        self.completed_handshake = False
         self.service_id = None
+        self.sock_info = None
         self.handled = False
 
     def contribute_socket(self, sock_info):
         """Provide socket information to the error handler."""
         self.max_wire_version = sock_info.max_wire_version
         self.sock_generation = sock_info.generation
         self.service_id = sock_info.service_id
-        self.completed_handshake = True
+        self.sock_info = sock_info
```

Comment:
contribute_socket is only ever called with an established connection (ie when `sock_info.completed_handshake == True`). So is this change needed at all?

Reply:
Oh interesting, let me try that

Reply:
Yeah that works